### ISSUE TITLE
[AVM Module Issue]: Error `ManagedEnvironmentInvalidNetworkConfiguration` when creating internal Azure Container App Environment

### ISSUE THREAD

**Issue by timur-khadimullin** [Azure/terraform-azurerm-avm-res-network-virtualnetwork#101]
### Check for previous/existing GitHub issues

- [X] I have checked for previous/existing GitHub issues

### Issue Type?

Bug

### Description

Consider the following terraform file:

```HCL
## Variables
variable "project" {
  description = "Identifies project(s) that the resource supports"
  type        = string
  default     = "aca-vnet-integration-test"
}

variable "azure_region" {
  description = "Azure region"
  type        = string
  default     = "australiaeast"
}

variable "tenant_id" {
  description = "Azure tenant ID"
  type        = string
  default     = "xxx-xxx-xxx-xxx-xxx"
}

## Providers
provider "azurerm" {
  features {}
  tenant_id       = var.tenant_id
  subscription_id = "xxx-xxx-xxx-xxx-xxx"
}

## Versions

terraform {
  required_version = ">= 1.6.0"

  required_providers {
    azurerm = {
      source  = "hashicorp/azurerm"
      version = "~> 3.0"
    }
  }
}

## RG
resource "azurerm_resource_group" "rg" {
  name     = "${var.project}-rg"
  location = var.azure_region
}

# VNET & two subnets, both delegated to Microsoft.App/environments
module "avm-res-network-virtualnetwork" {
  source = "Azure/avm-res-network-virtualnetwork/azurerm"

  address_space      = ["10.11.0.0/16"]
  location            = azurerm_resource_group.rg.location
  name                = "${var.project}-vnet"
  resource_group_name = azurerm_resource_group.rg.name
  enable_telemetry = false
  subnets = {
    "failing_subnet" = {
      name             = "avm-${var.project}-aca-subnet"
      address_prefixes = ["10.11.0.0/24"]
      default_outbound_access_enabled = true
      delegation = [{
        name = "fail_aca_delegation"
        service_delegation = {
          name    = "Microsoft.App/environments"
          actions = ["Microsoft.Network/virtualNetworks/subnets/join/action"]
        }
      }]
    }
  }
}

resource "azurerm_subnet" "working_subnet" {
  name                 = "azurerm-${var.project}-aca-subnet"
  resource_group_name  = azurerm_resource_group.rg.name
  virtual_network_name = module.avm-res-network-virtualnetwork.name
  address_prefixes     = ["10.11.1.0/24"]
  
  delegation {
    name = "working_aca_delegation"
    service_delegation {
      name    = "Microsoft.App/environments"
      actions = [ "Microsoft.Network/virtualNetworks/subnets/join/action" ]
    }
  }
}

## Azure Container Apps. One integrated into each subnet from above
resource "azurerm_container_app_environment" "failing_aca" {
  name                       = "${var.project}-aca-failing"
  location                   = azurerm_resource_group.rg.location
  resource_group_name        = azurerm_resource_group.rg.name

  infrastructure_resource_group_name = "${var.project}-aca-managed-failing-rg"
  infrastructure_subnet_id           = module.avm-res-network-virtualnetwork.subnets["failing_subnet"].resource_id
  internal_load_balancer_enabled = true

  workload_profile {
    name = "Consumption"
    workload_profile_type  = "Consumption"
    maximum_count = 3
    minimum_count = 0
  }
  zone_redundancy_enabled = true
}

resource "azurerm_container_app_environment" "working_aca" {
  name                       = "${var.project}-aca-working"
  location                   = azurerm_resource_group.rg.location
  resource_group_name        = azurerm_resource_group.rg.name

  infrastructure_resource_group_name = "${var.project}-aca-managed-working-rg"
  infrastructure_subnet_id           = azurerm_subnet.working_subnet.id
  internal_load_balancer_enabled = true

  workload_profile {
    name = "Consumption"
    workload_profile_type  = "Consumption"
    maximum_count = 3
    minimum_count = 0
  }
  zone_redundancy_enabled = true
}
```

when I run `terraform apply` I get the following error message :

```bash
│ Error: creating Managed Environment (Subscription: "xxx-xxx-xxx-xxx-xxx"
│ Resource Group Name: "timur-aca-test-rg"
│ Managed Environment Name: "timur-aca-test-aca-failing"): performing CreateOrUpdate: unexpected status 400 (400 Bad Request) with error: ManagedEnvironmentInvalidNetworkConfiguration: The environment network configuration is invalid: The subnet and its addressPrefix could not be found.
│ 
│   with azurerm_container_app_environment.failing_aca,
│   on main.tf line 87, in resource "azurerm_container_app_environment" "failing_aca":
│   87: resource "azurerm_container_app_environment" "failing_aca" {
│ 
│ creating Managed Environment (Subscription: "xxx-xxx-xxx-xxx-xxx"
│ Resource Group Name: "timur-aca-test-rg"
│ Managed Environment Name: "timur-aca-test-aca-failing"): performing CreateOrUpdate: unexpected status 400 (400 Bad Request) with error: ManagedEnvironmentInvalidNetworkConfiguration: The environment network configuration is invalid:
│ The subnet and its addressPrefix could not be found.
```

upon inspecting the two provisioned subnets with `az network vnet subnet show`, I see a few differences but one particularly caught my eye:

![image](https://github.com/Azure/terraform-azurerm-avm-res-network-virtualnetwork/assets/148298411/c20a125c-3ca2-44e4-8cf2-d3df6778ce8f)

I expect both subnets to be able to be used with ACA.

**Comment by piotrkurylak** [comment]
Yeah, I've came into this also. Could you please review this issue?

**Comment by jaredfholgate** [comment]
Released in 0.3.0. Use the `address_prefix` variable instead of the `address_prefixes` variable for cases like this.
